### PR TITLE
[m65] add via, new mcu gd32f303, stm32f401 weact support for encoder and led strip

### PR DESCRIPTION
**Pull request by alinelena**
general updates and new features as support for new mcus via support and a monstrous unicode keymap

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/m65/rev2/config.h [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+#define RGB_DI_PIN  B15
+
+#define UNUSED_PINS
+
+#define ENCODER_RESOLUTION 4
+
+
+#undef RGBLED_NUM
+#define RGBLED_NUM 20
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/m65/rev2/config.h:44]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/m65/rev3/config.h [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+#define RGB_DI_PIN  B15
+
+#define UNUSED_PINS
+
+#define ENCODER_RESOLUTION 4
+
+
+#undef RGBLED_NUM
+#define RGBLED_NUM 20
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/m65/rev3/config.h:44]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/m65/rev2/board.h [hunk 3]
```diff
@@ -0,0 +1,4 @@
+#pragma once
```
[the remark this answers is not in the document]
Comment (alinelena) [reply keyboards/m65/rev2/board.h:17]: thanks, done!

File: keyboards/m65/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+#endif
+}
+
+bool led_update_user(led_t led_state) {
+    // Disable the default LED update code, so that lock LEDs could be reused to show layer status.
+    return false;
+}
+
+void matrix_scan_user(void) {
```
Comment (drashna) [inline keyboards/m65/keymaps/via/keymap.c:88]:
```suggestion
void matrix_slave_scan_user(void) { matrix_scan_user(); }
void matrix_scan_user(void) {
```

Comment (alinelena) [reply keyboards/m65/keymaps/via/keymap.c:88]: this is for split... that is request 14383 not this one

**Review by drashna — APPROVED** [review]

File: keyboards/mlego/m65/rev3/rules.mk [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+CONSOLE_ENABLE = no       # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+RGBLIGHT_DRIVER = WS2812
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (fauxpark) [inline keyboards/mlego/m65/rev3/rules.mk:22]:
```suggestion
```

File: keyboards/mlego/m65/rev3/rules.mk [hunk 6]
```diff
@@ -0,0 +1,26 @@
+# MCU name
+MCU = STM32F401  #STM32F401 weact
+
+# Bootloader selection
+BOOTLOADER = stm32-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes     # Virtual DIP switch configuration
```
Comment (fauxpark) [inline keyboards/mlego/m65/rev3/rules.mk:10]:
```suggestion
BOOTMAGIC_ENABLE = yes     # Enable Bootmagic Lite
```

File: keyboards/mlego/m65/rev3/rules.mk [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+RGBLIGHT_DRIVER = WS2812
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+ENCODER_ENABLE = yes        # Enable encoder
+
+
```
Comment (fauxpark) [inline keyboards/mlego/m65/rev3/rules.mk:26]:
```suggestion
```

File: keyboards/mlego/m65/rev3/rules.mk [hunk 8]
```diff
@@ -0,0 +1,26 @@
+# MCU name
+MCU = STM32F401  #STM32F401 weact
```
Comment (fauxpark) [inline keyboards/mlego/m65/rev3/rules.mk:2]:
```suggestion
MCU = STM32F401
```

File: keyboards/mlego/m65/rev2/rules.mk [hunk 9]
```diff
@@ -10,12 +10,17 @@ BOOTLOADER = stm32duino
...
 NKRO_ENABLE = yes           # USB Nkey Rollover
 BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
-RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+RGBLIGHT_DRIVER = WS2812
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
 AUDIO_ENABLE = no           # Audio output
+ENCODER_ENABLE = yes     # Enable encoder
+
+
```
Comment (fauxpark) [inline keyboards/mlego/m65/rev2/rules.mk:26]:
```suggestion
```

File: keyboards/mlego/m65/rev2/rules.mk [hunk 10]
```diff
@@ -10,12 +10,17 @@ BOOTLOADER = stm32duino
 BOOTMAGIC_ENABLE = yes     # Virtual DIP switch configuration
```
Comment (fauxpark) [inline keyboards/mlego/m65/rev2/rules.mk:10]:
```suggestion
BOOTMAGIC_ENABLE = yes     # Enable Bootmagic Lite
```

File: keyboards/mlego/m65/rev2/rules.mk [hunk 11]
```diff
@@ -10,12 +10,17 @@ BOOTLOADER = stm32duino
...
 MOUSEKEY_ENABLE = yes       # Mouse keys
 EXTRAKEY_ENABLE = yes       # Audio control and System control
-CONSOLE_ENABLE = yes        # Console for debug
+CONSOLE_ENABLE = yes       # Console for debug
```
Comment (fauxpark) [inline keyboards/mlego/m65/rev2/rules.mk:13]:
```suggestion
CONSOLE_ENABLE = yes        # Console for debug
```

File: keyboards/mlego/m65/rev2/rules.mk [hunk 12]
```diff
@@ -10,12 +10,17 @@ BOOTLOADER = stm32duino
...
 COMMAND_ENABLE = yes        # Commands for debug and configuration
 # Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
 SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
 # if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
 NKRO_ENABLE = yes           # USB Nkey Rollover
 BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
-RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+RGBLIGHT_DRIVER = WS2812
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (fauxpark) [inline keyboards/mlego/m65/rev2/rules.mk:22]:
```suggestion
```

File: keyboards/mlego/m65/rev1/rules.mk [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+CONSOLE_ENABLE = no        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+RGBLIGHT_DRIVER = WS2812
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (fauxpark) [inline keyboards/mlego/m65/rev1/rules.mk:22]:
```suggestion
```

File: keyboards/mlego/m65/rev1/rules.mk [hunk 14]
```diff
@@ -0,0 +1,24 @@
+# MCU name
+MCU = STM32F103
+
+# Bootloader selection
+BOOTLOADER = stm32duino
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes     # Virtual DIP switch configuration
```
Comment (fauxpark) [inline keyboards/mlego/m65/rev1/rules.mk:10]:
```suggestion
BOOTMAGIC_ENABLE = yes     # Enable Bootmagic Lite
```

File: keyboards/mlego/m65/keymaps/via/config.h [hunk 15]
```diff
@@ -18,3 +18,4 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #pragma once
 
 #define TAPPING_TOGGLE 2
+
```
Comment (fauxpark) [inline keyboards/mlego/m65/keymaps/via/config.h:21]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/mlego/m65/rev1/rules.mk [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
```
Comment (fauxpark) [inline keyboards/mlego/m65/rev1/rules.mk:18]:
```suggestion
NKRO_ENABLE = yes           # Enable N-Key Rollover
```

File: keyboards/mlego/m65/rev3/rules.mk [hunk 17]
```diff
@@ -0,0 +1,22 @@
...
+#
+BOOTMAGIC_ENABLE = yes     # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no       # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
```
Comment (fauxpark) [inline keyboards/mlego/m65/rev3/rules.mk:18]:
```suggestion
NKRO_ENABLE = yes           # Enable N-Key Rollover
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/mlego/m65/info.json [hunk 18]
```diff
@@ -0,0 +1,81 @@
...
+                {"x": 2, "y": 4},
+                {"x": 3, "y": 4},
+                {"x": 4, "y": 4},
+                {"x": 5, "y": 4},
+                {"x": 6, "y": 4},
+                {"x": 7, "y": 4},
+                {"x": 8, "y": 4},
+                {"x": 9, "y": 4},
+                {"x": 10, "y": 4},
+                {"x": 11, "y": 4}
```
Comment (noroadsleft) [inline keyboards/mlego/m65/info.json:76]: Missing comma -> syntax error

```suggestion
                {"x": 11, "y": 4},
```

File: keyboards/mlego/m65/m65.h [hunk 19]
```diff
@@ -0,0 +1,146 @@
...
+    {                                                                                                                                                                                                                                                                                                                                                                                                                           \
+        {K000, K001, K002, K003, K004, K005, K006, K007, K008, K009, K010, K011, K012}, {K100, K101, K102, K103, K104, K105, K106, K107, K108, K109, K110, K111, K112}, {K200, K201, K202, K203, K204, K205, K206, K207, K208, K209, K210, K211, K212}, {K300, K301, K302, K303, K304, K305, K306, K307, K308, K309, K310, K311, K312}, { K400, K401, K402, K403, K404, K405, K406, K407, K408, K409, K410, K411, K412 }        \
+    }
```
Comment (noroadsleft) [inline keyboards/mlego/m65/m65.h:25]:
```suggestion
#define LAYOUT_ortho_5x13( \
    K000, K001, K002, K003, K004, K005, K006, K007, K008, K009, K010, K011, K012, \
    K100, K101, K102, K103, K104, K105, K106, K107, K108, K109, K110, K111, K112, \
    K200, K201, K202, K203, K204, K205, K206, K207, K208, K209, K210, K211, K212, \
    K300, K301, K302, K303, K304, K305, K306, K307, K308, K309, K310, K311, K312, \
    K400, K401, K402, K403, K404, K405, K406, K407, K408, K409, K410, K411, K412  \
) { \
    { K000, K001, K002, K003, K004, K005, K006, K007, K008, K009, K010, K011, K012 }, \
    { K100, K101, K102, K103, K104, K105, K106, K107, K108, K109, K110, K111, K112 }, \
    { K200, K201, K202, K203, K204, K205, K206, K207, K208, K209, K210, K211, K212 }, \
    { K300, K301, K302, K303, K304, K305, K306, K307, K308, K309, K310, K311, K312 }, \
    { K400, K401, K402, K403, K404, K405, K406, K407, K408, K409, K410, K411, K412 }  \
}
```

File: keyboards/mlego/m65/m65.h [hunk 20]
```diff
@@ -0,0 +1,146 @@
...
+
+enum layer_names { _QW = 0, _LWR, _RSE, _ADJ };
+
```
Comment (noroadsleft) [inline keyboards/mlego/m65/m65.h:28]:
```suggestion
```

File: keyboards/mlego/m65/keymaps/default/keymap.c [hunk 21]
```diff
@@ -0,0 +1,137 @@
...
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include QMK_KEYBOARD_H
+
```
Comment (noroadsleft) [inline keyboards/mlego/m65/keymaps/default/keymap.c:19]:
```suggestion

enum layer_names {
    _QW = 0,
    _LWR,
    _RSE,
    _ADJ
};

```

File: keyboards/mlego/m65/keymaps/dk/keymap.c [hunk 22]
```diff
@@ -0,0 +1,136 @@
...
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include QMK_KEYBOARD_H
+#include "keymap_danish.h"
+
```
Comment (noroadsleft) [inline keyboards/mlego/m65/keymaps/dk/keymap.c:20]:
```suggestion

enum layer_names {
    _QW = 0,
    _LWR,
    _RSE,
    _ADJ
};

```

File: keyboards/mlego/m65/keymaps/uk/keymap.c [hunk 23]
```diff
@@ -0,0 +1,217 @@
...
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include QMK_KEYBOARD_H
+#include "keymap_uk.h"
+
```
Comment (noroadsleft) [inline keyboards/mlego/m65/keymaps/uk/keymap.c:20]:
```suggestion

enum layer_names {
    _QW = 0,
    _LWR,
    _RSE,
    _ADJ
};

```

File: keyboards/mlego/m65/keymaps/via/keymap.c [hunk 24]
```diff
@@ -0,0 +1,134 @@
...
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include QMK_KEYBOARD_H
+
```
Comment (noroadsleft) [inline keyboards/mlego/m65/keymaps/via/keymap.c:19]:
```suggestion

enum layer_names {
    _QW = 0,
    _LWR,
    _RSE,
    _ADJ
};

```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Review by noroadsleft — APPROVED** [review]
Looks good to me.

**Pull request merged**